pkg/stub/handler.go

```diff
@@ -186,64 +183,91 @@ func (h *Handler) processImageStreamWatchEvent(is *imagev1.ImageStream) error {
 			}
 
 			logrus.Debugf("pending is %v for %s", pending, is.Name)
+			// with the mutex gone, we started getting imagestream complete situations
+			// before the initial startup run completed all the imagestreams and updated
+			// the contition to true; so now, if in progress is false, but exists is also
+			// false, meaning this is part of the initial startup, we throw the error to
+			// so we process the pending condition after the startup completed
+			processing := srcfg.Condition(v1alpha1.ImageChangesInProgress)
+			exists := srcfg.Condition(v1alpha1.SamplesExist)
+			if !pending &&
+				processing.Status != corev1.ConditionTrue &&
+				exists.Status != corev1.ConditionTrue {
+				return fmt.Errorf("imagestream %s is no longer pending but creation loop still in progress", is.Name)
+			}
```

Comment:
Can we simplify this down to "ignore imagestream events as long as SamplesExist is false"?

And when we start creating samples, we set ImageChangeInProgress to true?

Reply:
To answer you second question, yes, we set in progress to true when we start creating samples ... for example https://github.com/openshift/cluster-samples-operator/blob/master/pkg/stub/handler.go#L1065

To you first question, yeah I think you are right.  Change will be part of next push, unless something unexpected happens during testing.